### ISSUE TITLE
Make CodeCoverage::initializeData() public to avoid wrong coverage report generation

### ISSUE THREAD

**Issue by turgutsaricam** [sebastianbergmann/php-code-coverage#685]
| Q                         | A
| --------------------------| ---------------
| php-code-coverage version | 7.0.6
| PHP version               | 7.2.19
| Driver                    | Xdebug
| Xdebug version (if used)  | 2.6.0
| Installation Method       | Composer
| Usage Method              | Manual report generation
| PHPUnit version (if used) | Not applicable

I needed to cover browser tests that are run by using Selenium. To create coverage report, I prepended a PHP file that is run before an HTTP request is processed so that I can start coverage and generate a report when the tests are done. However, using `CodeCoverage` class to create coverage data when running browser tests takes too much time, making the tests run about 10 times slower. Hence, I used Xdebug's `xdebug_get_code_coverage()` function to retrieve the coverage data and save it. This is way faster than doing it with `CodeCoverage::start()`. There is no problem until here.

The problem arises when I try to generate a report by creating a `CodeCoverage` instance and using `append()` method to add all coverage data that were dumped before. You can observe the PHP code below to see how I generate a combined report using previously saved coverage data.

```php
// Create a CodeCoverage from dumped coverage files

// Get dumped coverage file paths. These are the files that are created by first calling
//      xdebug_start_code_coverage(XDEBUG_CC_UNUSED | XDEBUG_CC_DEAD_CODE)
// 
// Then, after the tests are finished, calling
//      xdebug_stop_code_coverage(false);
//
// And then saving the coverage data via
//      file_put_contents('path/to/coverage-testName.json', json_encode(xdebug_get_code_coverage()))
$coverages = glob("coverages/*.json");

$finalCoverage = new SebastianBergmann\CodeCoverage\CodeCoverage();
$finalCoverage->setAddUncoveredFilesFromWhitelist(true);
$finalCoverage->setProcessUncoveredFilesFromWhitelist(true);
$finalCoverage->setCheckForUnexecutedCoveredCode(true);

$filter = $finalCoverage->filter();

$appDir = '/path/to/app/';
$filter->addDirectoryToWhitelist($appDir);
$filter->removeDirectoryFromWhitelist($appDir . 'vendor');
$filter->removeDirectoryFromWhitelist($appDir . 'storage');
$filter->removeDirectoryFromWhitelist($appDir . 'views');

// The data must be initialized to produce a proper coverage report.
$finalCoverage->initializeData();

// Append each coverage data to the CodeCoverage instance
foreach ($coverages as $coverageFile) {
    $testName = str_ireplace("coverage-", "", basename($coverageFile, ".json"));

    $codeCoverageData = json_decode(file_get_contents($coverageFile), JSON_OBJECT_AS_ARRAY);
    $finalCoverage->append($codeCoverageData, $testName);
}

// Generate an HTML report
$report = new \SebastianBergmann\CodeCoverage\Report\Html\Facade();
$report->process($finalCoverage, "reports/html");
```

You can see that I call `initializeData()` method. The reason for doing so is that the coverage report does not contain uncovered lines if `initializeData()` method is not called. The wrong coverage report shows 100% coverage, although coverage is not 100%. The following report is retrieved by *not* calling `initializeData()`:

![ss-coverage-1-no-initialization](https://user-images.githubusercontent.com/5666595/60919151-b57dda80-a29d-11e9-9fd2-8b9e99b16c03.jpg)

The report below is retrieved by calling `initializeData()`:

![ss-coverage-2-with-initialization](https://user-images.githubusercontent.com/5666595/60919183-cb8b9b00-a29d-11e9-911b-6351669e213e.jpg)

This is an example method from the report generated by *not* calling `initializeData()`:

![ss-code-1-no-initialization](https://user-images.githubusercontent.com/5666595/60919250-eeb64a80-a29d-11e9-9750-418493486525.jpg)

The following one is the same method from the report generated by calling `initializeData()`:

![ss-code-2-with-initialization](https://user-images.githubusercontent.com/5666595/60919280-01c91a80-a29e-11e9-8c32-12c5b6be4d64.jpg)

As you can see, when the method is called, the report is correctly generated.

The problem is the method's visibility is `private`. Therefore, it is not possible to call the method outside of the class. Could you please change the visibility of `CodeCoverage::initializeData()` to `public`?

**Comment by sebastianbergmann** [comment]
The method is `private` because it is a private implementation detail. You are trying to use this component in a way that is not intended. I am sorry to say this but when you do not want to use this component's abstraction over Xdebug, for instance, then you are on your own.

**Comment by turgutsaricam** [comment]
@sebastianbergmann Thanks, you are definitely right but I took my chances since the implementation is already there. But, your reply was actually helpful since it made me see another way that I could not see before. It is possible to call `initializeData()` indirectly, by calling `start()`. So, I will just leave my solution here if anyone is interested in.

Instead of calling
```php
$finalCoverage->initializeData();
```

I just call
```php
$finalCoverage->start('some-id');
$finalCoverage->stop();
```

I did not see any side effects, since there is no code run in between `start` and `stop`.
